src/Factories/FunFunFactory.php

```diff
@@ -1372,4 +1373,11 @@ public function setContentProvider( ContentProvider $contentProvider ): void {
 	private function getContentProvider(): ContentProvider {
 		return $this->pimple['content_provider'];
 	}
+
+	public function newMailTemplateFilenameTraversable(): \Traversable {
```

Comment:
There is a cost to being more specific here. If you hint it as Traversable, then the consumer knows it should just be used as a Traversable. If you hint as the specific class, which IMO is an implementation detail, then the consumer needs to first figure out that it is just a Traversable and nothing more is going on. In this case it does not matter much either way... if it was a factory in a library or similar situation it'd be much more important.

The rule I've been following is use for \Foo\Bar but not for \Foo, though with doing it anyway for \Foo being fine. Example: our tests used to have `extends \PHPUnit_Framework_TestCase`